tests/Mantis/MentionParsingTest.php

```diff
@@ -0,0 +1,113 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Test cases for mention API
+ *
+ * @package    Tests
+ * @subpackage UnitTests
+ * @copyright Copyright 2002-2016  MantisBT Team - mantisbt-dev@lists.sourceforge.net
+ * @link http://www.mantisbt.org
+ */
+
+/**
+ * Includes
+ */
+require_once dirname( dirname( __FILE__ ) ) . '/TestConfig.php';
+
+require_mantis_core();
+
+require_api( 'mention_api.php' );
+
+/**
+ * Test cases for parsing functionality in mention API.
+ * @package    Tests
+ * @subpackage Mention
+ */
+class MentionParsingTest extends PHPUnit_Framework_TestCase {
+	public function testNoMention() {
+		$this->checkMentions( 'some random string.', array() );
+	}
+
+	public function testNoMentionWithAtSign() {
+		$this->checkMentions( 'some random string with @ sign.', array() );
+	}
+
+	public function testNoMentionWithMultipleAtSigns() {
+		$this->checkMentions( 'some random string with @@vboctor sign.', array() );
+	}
+
+	public function testJustMention() {
+		$this->checkMentions( '@vboctor', array( 'vboctor' ) );
+	}
+
+	public function testWithDotInMiddle() {
+		$this->checkMentions( '@victor.boctor', array( 'victor.boctor' ) );
+	}
```

Comment:
Add test to email address format
Eg:
xxxx@vboctor
xxxx@victor.boctor
should not be matched as user mention